### PR TITLE
[WIP] Move JSON/XML modules to separate projects

### PR DESCRIPTION
**Pull request by mielientiev**
The main goal of this PR is to move JSON/XML modules to separate projects.

- [x] remove Data module dependency from JSON modules
- [x] remove Data module dependency from XML modules
- [ ] do cleanup
- [ ] fix most common XML/JSON bugs
     - [x] XML: https://github.com/propensive/rapture/issues/262
     - [x] XML: https://github.com/propensive/rapture/issues/263
     - [ ] XML: https://github.com/propensive/rapture/issues/264
     - [ ] XML: https://github.com/propensive/rapture/issues/253
     - [ ] XML: https://github.com/propensive/rapture/issues/250
     - [ ] XML: https://github.com/propensive/rapture/issues/247
     - [ ] XML/JSON: https://github.com/propensive/rapture/issues/238 
- [ ] add more XML/JSON unit tests
- [ ] switch to contextual
- [ ] Move JSON/XML modules to separate repos

### CODE REVIEW

File: json/shared/src/main/scala/rapture/json/macros.scala [hunk 1]
```diff
@@ -34,3 +33,246 @@ private[json] object JsonMacros {
...
+          mode.unwrap($deref.as($inferredImplicit, mode.generic), ${"." + p.name.decodedName})
+        """
+      }
+
+      if (implicitSearchFailures.nonEmpty) {
+        val paramStrings = implicitSearchFailures map {
+          case (t, p1 :: Nil) =>
+            s"parameter `$p1' of type $t"
+          case (t, p1 :: ps) =>
+            s"parameters ${ps.map(v => s"`$v'").mkString(", ")} and `$p1' of type $t"
```
Comment (propensive) [inline json/shared/src/main/scala/rapture/json/macros.scala:182]: Is this red marking just GitHub's parser getting confused?

**Review by propensive — COMMENTED** [review]
Just to confirm, regarding all the "new" code that is added, is that simply copy/pasted from `rapture.data`, then fixed to correct references to `rapture.data`?

I'm assuming that the deletions of the original files will be made after you've done the same thing with Rapture XML...

Anyhow, it looks good to continue now! Let me know if you need any help/explanations.

File: json/shared/src/main/scala/rapture/json/macros.scala (hunk 1 above)
Comment (mielientiev) [reply json/shared/src/main/scala/rapture/json/macros.scala:182]: I think so. All code compiles and tests passes

**Comment by mielientiev** [comment]
@propensive Yep. For now, it's just copy-pasting from data module to json/xml module.
In future, all json/xml modules will only depend on `rapture-core`. 
after I've done the same thing with Rapture XML, we can move Rapture JSON & XML to the separate repos

**Comment by propensive** [comment]
I'm going to tidy up Rapture, and keep track of this in Xylophone from now on.

**Pull request closed without merging**